libexec/rtld-elf/aarch64/rtld_start.S

```diff
@@ -185,7 +166,16 @@ ENTRY(C18N_SYM(_rtld_bind_start))
 	ldr	PTR(0), [PTR(16), #-PTR_WIDTH]
 
 	/* Call into rtld */
+#ifdef CHERI_LIB_C18N
+	ldr	c10, c18n_rtld_bind
```

Comment:
Why not just always do an indirect call for these? Then you can drop C18N_VARIANT entirely and have one single set of assembly functions.